File: src/displays/environment.ts
import { blue } from "@std/fmt/colors";
import type { Logger } from "./../core/logger.ts";

export default async (
  { homedir, dotfilesFilesDir, logger, color }: {
    homedir: string;
    logger: Logger;
    dotfilesFilesDir: string;
    color?: boolean;
  },
) => {
  await logger.log.log(`
-----------------------
Environment:

HOME:               ${color ? blue(homedir) : homedir}
DOTFILES FILES DIR: ${color ? blue(dotfilesFilesDir) : dotfilesFilesDir}
-----------------------
  `.trim());
};

File: src/displays/environment.go
package displays

import (
	"strings"

	"github.com/fatih/color"
	"github.com/m4rc3l05/dots/src/core"
)

func PrintEnvironment(
	homedir string,
	dotfilesFilesDir string,
	logger core.ILogger,
) {
	logger.Lognl(strings.TrimSpace(`
-----------------------
Environment:

HOME:               %s
DOTFILES FILES DIR: %s
-----------------------
`), color.BlueString(homedir), color.BlueString(dotfilesFilesDir))
}
